### PR TITLE
Reduce with sum

### PR DESCRIPTION
**Pull request by marc9595**
I use `pa.Array.sum()` in `self._reduce()` to implement the case where `name == "sum"` and `name=="mean"` for numeric type. The code is pretty straightforward. 
I also put some new tests and removed the initial ones in `test_pandas_extension.py` for `"sum"`and `"mean"` because they should be skipped if those cases are implemented in `reduce` for numeric types.

### CODE REVIEW

File: fletcher/base.py [hunk 1]
```diff
@@ -285,8 +285,16 @@ def _reduce(self, name, skipna=True, **kwargs):
             return any_op(self.data, skipna=skipna)
         elif name == "all" and pa.types.is_boolean(self.dtype.arrow_dtype):
             return all_op(self.data, skipna=skipna)
-
-        raise TypeError(f"cannot perform {name} with type {self.dtype}")
+        elif name == "sum" and (
+            pa.types.is_integer(self.data.type) or pa.types.is_floating(self.data.type)
```
Comment (artemru) [inline fletcher/base.py]: you could pyarrow raise an error

Comment (marc9595) [reply fletcher/base.py]: Done. You're right, this would make sure that we handle correctly all the different types.

File: tests/conftest.py [hunk 2]
```diff
@@ -64,10 +64,10 @@ def compare_operators_no_eq_ne(request):
 
 
 _all_numeric_reductions = [
-    "sum",
+    # "sum",
     "max",
     "min",
-    "mean",
+    # "mean",
```
Comment (krivonogov) [inline tests/conftest.py]: why commenting this out?

Comment (krivonogov) [reply tests/conftest.py]: ah ok, I see in description 👌

**Comment by marc9595** [comment]
I am merging. Thanks !

**Pull request merged**